core/src/main/scala/org/apache/spark/NewAccumulator.scala

```diff
@@ -209,9 +210,9 @@ private[spark] object AccumulatorContext {
    */
   def newId(): Long = nextId.getAndIncrement
 
-  def numAccums: Int = synchronized(originals.size)
+  def numAccums: Int = originals.size
 
-  def accumIds: Set[Long] = synchronized(originals.keySet().asScala.toSet)
+  def accumIds: Set[Long] = originals.keySet().asScala.toSet
```

Comment:
I rewrote the assertions in InternalAccumulatorSuite
I think accumIds is not needed.

Once @cloud-fan confirms, I will remove accumIds.